tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -0,0 +1,383 @@
+            }
+
+            // Set OTEL status code and OTEL status description
+            if (span.GetTag("otel.status_code") is null)
+            {
+                span.SetTag("otel.status_code", "STATUS_CODE_UNSET");
+            }
+
+            // Map the OTEL status to error tags
+            AgentStatus2Error(activity, span);
+
+            if (span.OperationName is null)
+            {
+                if (activity5 is not null)
+                {
+                    // Later: Support config 'span_name_as_resource_name'
+                    // Later: Support config 'span_name_remappings'
+                    span.OperationName = activity5.Source.Name switch
+                    {
+                        string libName when !string.IsNullOrEmpty(libName) => $"{libName}.{SpanKindNames[(int)activity5.Kind]}",
+                        _ => $"opentelemetry.{SpanKindNames[(int)activity5.Kind]}",
+                    };
+                }
+                else
+                {
+                    span.OperationName = activity.OperationName;
+                }
+            }
+
+            // Update Service with a reasonable default
+            if (span.ServiceName is null)
+            {
+                span.ServiceName = span.GetTag("peer.service") switch
+                {
+                    string peerService when !string.IsNullOrEmpty(peerService) => peerService,
+                    _ => "OTLPResourceNoServiceName",
+                };
+            }
+
+            // Update Resource with a reasonable default
+            if (span.ResourceName is null)
+            {
+                // TODO: Implement resourceFromTags
+                // See: https://github.com/DataDog/datadog-agent/blob/67c353cff1a6a275d7ce40059aad30fc6a3a0bc1/pkg/trace/api/otlp.go#L555
+
+                // Fallback: Use the information provided by Activity
+                if (activity5 is not null)
+                {
+                    span.ResourceName = activity5.DisplayName;
+                }
+                else
+                {
+                    span.ResourceName = activity.OperationName;
+                }
+            }
+
+            // Update Type with a reasonable default
+            if (string.IsNullOrWhiteSpace(span.Type))
+            {
+                span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
+            }
+        }
+
+        internal static void SetTagObject(Span span, string key, object? value)
+        {
+            if (value is null)
+            {
+                span.SetTag(key, null);
+                return;
+            }
+
+            switch (value)
+            {
+                case char c: // TODO: Can't get here from OTEL API, test with Activity API
+                    AgentSetOtlpTag(span, key, c.ToString());
+                    break;
+                case string s:
+                    AgentSetOtlpTag(span, key, s);
```

Comment:
We have to support other types because the OTEL API (and Activity API) allow adding values of type `object`. This logic follows the trace agent logic of getting the type of the attribute and placing it into `metrics` for numerical values and `meta` for everything else

Reply:
Ok, thanks.